File: services/backend/src/main/java/com/gume/sonar/domain/Report.java
package com.gume.sonar.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Report {
    private UUID id;
    private String title;
    private User user;
    private String client;
    private String analysis;
    private String transcript;
    private LocalDateTime creationDate;
}

File: services/frontend/src/domain/models/Report.ts
import { type User, type Client } from './Client';

export type ReportStatus = 'PROCESSING' | 'COMPLETED';

export interface Report {
  id: string;
  title: string;
  user: User;
  client: Client;
  analysis?: string;
  transcript?: string;
  creationDate: string;
  status: ReportStatus;
}
